homeassistant/components/schlage/entity.py

```diff
@@ -30,17 +31,19 @@ def __init__(
         )
 
     @property
-    def _lock_data(self) -> LockData:
+    def _lock_data(self) -> LockData | None:
         """Fetch the LockData from our coordinator."""
-        return self.coordinator.data.locks[self.device_id]
+        return self.coordinator.data.locks.get(self.device_id, None)
 
     @property
-    def _lock(self) -> Lock:
+    def _lock(self) -> Lock | None:
         """Fetch the Schlage lock from our coordinator."""
+        if self._lock_data is None:
+            return None
         return self._lock_data.lock
 
     @property
     def available(self) -> bool:
         """Return if entity is available."""
         # When is_locked is None the lock is unavailable.
-        return super().available and self._lock.is_locked is not None
+        return super().available and getattr(self._lock, "is_locked", None) is not None
```

Comment:
Why don't you refactor available to be 
```py
return super().available and self.device_id in self.coordinator.data.locks
```
It would render the entities as unavailable instead

Reply:
You can still have a bad data poll that renders the lock unavailable. (pesky cloud services...)

I refactored this to not use getattr, which I think is still better.